cc/dual_net/reloading_dual_net.cc

```diff
@@ -0,0 +1,194 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cc/dual_net/reloading_dual_net.h"
+
+#include <cstdio>
+#include <utility>
+
+#include "absl/memory/memory.h"
+#include "absl/time/clock.h"
+#include "cc/file/path.h"
+#include "cc/file/utils.h"
+
+namespace minigo {
+
+// ReloadingDualNetUpdater methods follow.
+
+ReloadingDualNetUpdater::ReloadingDualNetUpdater(const std::string& pattern,
+                                                 DualNetFactory* factory_impl)
+    : factory_impl_(factory_impl) {
+  auto pair = file::SplitPath(pattern);
+
+  directory_ = std::string(pair.first);
+  MG_CHECK(directory_.find('%') == std::string::npos &&
```

Comment:
worth it to extract the validator to an independent method?

Reply:
might allow for testability for any globby/stringy code

Reply:
good point. added some tests